src/main/java/graphql/analysis/MaxQueryComplexityInstrumentation.java

```diff
@@ -54,16 +54,19 @@ public MaxQueryComplexityInstrumentation(int maxComplexity, FieldComplexityCalcu
             }
             QueryTraversal queryTraversal = newQueryTraversal(parameters);
 
-            Map<QueryVisitorEnvironment, List<Integer>> valuesByParent = new LinkedHashMap<>();
-            queryTraversal.visitPostOrder(env -> {
-                int childsComplexity = 0;
-                QueryVisitorEnvironment thisNodeAsParent = new QueryVisitorEnvironment(env.getField(), env.getFieldDefinition(), env.getParentType(), env.getParentEnvironment(), env.getArguments());
-                if (valuesByParent.containsKey(thisNodeAsParent)) {
-                    childsComplexity = valuesByParent.get(thisNodeAsParent).stream().mapToInt(Integer::intValue).sum();
+            Map<QueryVisitorFieldEnvironment, List<Integer>> valuesByParent = new LinkedHashMap<>();
```

Comment:
Maybe put a builder on QueryVisitorFieldEnvironment.  It will allow it to evolve more easily over time.

Reply:
I extracted an Interface instead.